src/TMap.cpp

```diff
@@ -1222,32 +1225,35 @@ bool TMap::serialize(QDataStream& ofs, int saveVersion)
 
     if (mSaveVersion < 21) {
         // Before version 21 the map labels were stored within this class:
-        // number of labels per area - we need this as there is no delimiter
-        // between each area's map labels
-        int areasWithLabels = 0;
+        // First we have the number of labels per area - we need this as there
+        // is no delimiter between each area's map labels
+        QMap<int, TArea*> areasWithPermanentLabels;
         // Need to count the areas that have mapLabels:
-        for (const auto pArea : mpRoomDB->getAreaPtrList()) {
-            if (pArea && !pArea->mMapLabels.isEmpty()) {
-                ++areasWithLabels;
-            }
-        }
-        ofs << areasWithLabels;
         QMapIterator<int, TArea*> itArea(mpRoomDB->getAreaMap());
-        while (itArea.hasNext()) {
+        while (itArea.hasNext()){
+            // Now we have temporary labels we need to identify areas with
+            // perminent ones:
```

Comment:
```suggestion
            // permanent ones:
```

Not a blocker of course! :)

Reply:
Actually that spelling mistake appears three times in the code base as I see it now - it is just I do not have the same spell-checking facility in my (currently running OS - Windoze) development environment... *\*sigh\**

Let me throw that fix into another commit.